### ISSUE TITLE
Game Won't Read Player Models Inside PAK/ZIP

### ISSUE THREAD

**Issue by deuxsonic** [yquake2/yquake2#119]
Another suggested feature, but would it be possible to allow the game to read player model data inside of a PAK or ZIP file? It seems that is the only game data it cannot read out of archives.

**Comment by mckaygerhard** [comment]
i think obviously its a ruindows problems, must revise what are chappened when loading process are executed, i have the pak7.pak with the hires models textures and it loaded... 

i clarely see:
```
Added packfile '/home/remoto/Descargas/quake2/baseq2/pak7.pak' (88 files).
Added packfile '/home/remoto/Descargas/quake2/baseq2/pak8.pak' (122 files).
Added packfile '/home/remoto/Descargas/quake2/baseq2/z6_32bitclr_png_allmaps.pak' (1655 files).
```
as can see the 88 files can be loaded, 
so then u must revise the loading process.. of course using console and see what are happened

see the loading process|:
```
remoto@venenux:~$ /usr/lib/yamagi-quake2/quake2 +set basedir /home/remoto/Descargas/quake2

Yamagi Quake II v5.31
=====================

Client build options:
 - SDL2 (using 1.2)
 + CD audio
 + OGG/Vorbis
 + OpenAL audio
 + Zip file support
Platform: Linux
Architecture: i386
Byte ordering: little endian

Using /usr/share/games/quake2//baseq2/ to fetch paks
Using binary dir /usr/lib/yamagi-quake2/baseq2/ to fetch paks
Added packfile '/home/remoto/Descargas/quake2/baseq2/pak0.pak' (3307 files).
Added packfile '/home/remoto/Descargas/quake2/baseq2/pak1.pak' (279 files).
Added packfile '/home/remoto/Descargas/quake2/baseq2/pak2.pak' (2 files).
Added packfile '/home/remoto/Descargas/quake2/baseq2/pak7.pak' (88 files).
Added packfile '/home/remoto/Descargas/quake2/baseq2/pak8.pak' (122 files).
Added packfile '/home/remoto/Descargas/quake2/baseq2/z6_32bitclr_png_allmaps.pak' (1655 files).
Using '/home/remoto/.yq2/baseq2' for writing.
execing default.cfg
execing yq2.cfg
execing config.cfg
basedir is write protected.
Console initialized.
```

**Comment by mckaygerhard** [comment]
investigating i noted some mistakes:

my pak models textures have 88 files inside have many monster.. but some does not load .. but player load! must be some code related

maybe could not due yamagi are focused in classic not inproved.. but could be a great feature

**Comment by deuxsonic** [comment]
I'm specifically talking about the contents of the 'players' folder of the game. Remove it and place it inside a new pak file to demonstrate: the game is seemingly blind to it. If you try to customize your player it will say that no player models have been found.
 


    On Monday, November 23, 2015 3:09 PM, PICCORO Lenz McKAY <notifications@github.com> wrote:
 

 investigating i noted some mistakes:my pak models textures have 88 files inside have many monster.. but some does not load .. but player load! must be some code relatedmaybe could not due yamagi are focused in classic not inproved.. but could be a great feature—

**Comment by mckaygerhard** [comment]
umm i confirmed, ok that-s true, players models can be loaded from pak files.. it seems due the focused of yamagi-quake this can be a mayor feature request and maybe not so much possible...

**Comment by Quindorrian** [comment]
Well, yamagi-quake is able to read world and object textures from a peak file.  I've confirmed that it works.  I wonder why player models are the exception?...

**Comment by mckaygerhard** [comment]
well i not have at my home.. when i tested i used 5.30, so in few days i returned and test agains my gammer linux configuration and see if loaded or not in the lasted i see in debian (5.31) and also download and test in lasted (5.33), then report feedback

**Comment by Yamagi** [comment]
The problem is Quake IIs horrible architecture. About 50% of the code uses the virtual file system (filesystem.c) and the other 50% doesn't. The model loading is part of the latter. Fixing this is a huge project, very likely to introduce new hard to analyse bugs and breaking at least some mods:

- The virtual file system has to be refactored, bugfixed and optimized.
- All code needs to be altered to use the virtual file system.

This currently out of YQ2s scope. We are quite happy that after countless hours of work the game runs nearly bug free and we're very reluctant in doing any invasive changes that'll break things again.

**Comment by mckaygerhard** [comment]
well @yamagi has right! that means complety redone and will changhe the entire codebase

umm i think better its a script that makes symlinks of the files,, real files in lowercase and symlinks in uppercase, so then all cases will exists!

i'll try that method and then report feedback here! but not this month, due i'm very busy! in a 2 weeks i think!

**Comment by Yamagi** [comment]
This is fixed in out experimental 'version6' branch. Player models, cinematics and music can now be read from pak and zip files.

**Comment by mckaygerhard** [comment]
hello @Yamagi this xperimental branch will be merged soon or later in main?
and second, this xperimental branch still compile in older env (sdl 1.2, alsa older, old xorg < 1.5)

currently i can compile yamagi quake with sdl 1.2.12, alsa 1.0.18, and xorg < 1.5 and made run in low end and very very older machines like my asus eepc 2g surf (using debian lenny)

**Comment by Yamagi** [comment]
Hi,
the version6 branch will be merged to master and become YQ2 6.0. We're doing the work in a branch just in case that another 5.x bugfix release is needed. It this time the branch should be buildable with the same dependencies as master, e.g. SDL 1.2 is still suported. Hardware requirements are nearly the same, just some very early GPUs were dropped. The Nvidia TNT and everything newer should be okay.
